web/src/app/procurement/pending/hooks.ts

```diff
@@ -1,70 +1,336 @@
-		{ key: "vehicle", label: "Vehicle" },
-		{ key: "requestedBy", label: "Requested By" },
-		{ key: "comments", label: "Comments" },
-		{ key: "status", label: "Status" },
-		{ key: "actions", label: "Actions" },
+		{
+			key: "status",
+			label: "Status",
+			format: (value: StatusEnum) => {
+				const style = getStatusStyle(value);
+				return React.createElement("span", { style }, value);
+			},
+		},
+		{
+			key: "action",
+			label: "ACTION",
+			actions: [
+				{
+					name: "View",
+					handleClick: (row: RequestedDetailDto) => handleViewPendingItems(row),
+				},
+			],
+		},
 	];
 
-	const dataSource: any[] = [
+	const dataSource: RequestedDetailDto[] = requestedDetails?.data ?? [];
+
+	const columnSchemaForView: IColumnSchema<RequestedItemDto>[] = [
+		{ key: "requestCode", label: "Request Code" },
+		{ key: "itemName", label: "Item" },
+		{ key: "quantity", label: "Quantity" },
+		{ key: "unitsName", label: "Units" },
+		{ key: "brand", label: "Brand" },
+		{ key: "specification", label: "Specification" },
 		{
-			id: 1,
-			company: "Company A",
-			department: "HR",
-			category: "Office Supplies",
-			vehicle: "Car A",
-			requestedBy: "John Doe",
-			comments: "Urgent request",
-			status: "forAOC",
-			date: "2023-10-01",
+			key: "itemStatus",
+			label: "Status",
+			format: (value: StatusEnum) => {
+				const style = getStatusStyle(value);
+				return React.createElement("span", { style }, value);
+			},
 		},
+	];
+
+	const headerButton: IField<ButtonProps>[] = [
 		{
-			id: 2,
-			company: "Company B",
-			department: "Finance",
-			category: "IT Equipment",
-			vehicle: "Car B",
-			requestedBy: "Jane Smith",
-			comments: "Standard request",
-			status: "pending",
-			date: "2023-10-02",
+			fieldType: "button",
+			fieldProps: {
+				children: "Approved",
+				name: "approve",
+				color: "primary",
+				variant: "outlined",
+				sx: { width: 120 },
+				onClick: () => handleHeaderActions("approve"),
+			},
 		},
 		{
-			id: 3,
-			company: "Company C",
-			department: "Operations",
-			category: "Machinery",
-			vehicle: "Truck A",
-			requestedBy: "Alice Johnson",
-			comments: "High priority",
-			status: "received",
-			date: "2023-10-03",
+			fieldType: "button",
+			fieldProps: {
+				children: "Add Items",
+				name: "addItems",
+				color: "primary",
+				variant: "outlined",
+				sx: { width: 120 },
+				onClick: () => handleHeaderActions("addItems"),
+			},
+		},
+		{
+			fieldType: "button",
+			fieldProps: {
+				children: "Cancel Request",
+				name: "cancel",
+				color: "primary",
+				variant: "outlined",
+				sx: { width: 150 },
+				onClick: () => handleHeaderActions("cancel"),
+			},
 		},
 	];
 
-	const filteredDataSource = dataSource.filter(
-		(item) => item.status === "pending"
-	);
+	const handleAddItems = () => setOpen((state) => !state);
+
+	const handleSelectRow = (rows: RequestedDetailDto[]) => {
+		setSelectedRows(rows);
+	};
+
+	const handleHeaderActions = (type: ActionType) => {
```

Comment:
E Separate ni ang functions
```suggestion
	const handleHeaderActions = () => {
```